### ISSUE TITLE
Wrap code (at 80 chars) in sample code?

### ISSUE THREAD

**Issue by awalGarg** [mozilla/serviceworker-cookbook#108]
Heyo! Thanks for the really awesome cookbook :)

I was wondering if the code samples could be restricted to a specific line width, say 80 chars?

For example, https://serviceworke.rs/push-simple_index_doc.html introduces horizontal scrolling on my 1366x768 laptop screen.

This isn't a big issue at all, but *if* owners think it would be "okay" to have this, I'll be ready with a PR ;-)

<3, Awal

**Comment by marco-c** [comment]
Yes, Awal. I think it would be great! Lines are too long even for my 1920x1080 screen :smile: 

It'd be great if you could also use the Q-style for promises, which makes lines shorter.
For example, this:
```JavaScript
return registration.pushManager.subscribe({ userVisibleOnly: true }).then(function(newSubscription) {
```
would become:
```JavaScript
return registration.pushManager.subscribe({ userVisibleOnly: true })
.then(function(newSubscription) {
```

**Comment by mykmelez** [comment]
Even wrapping at 80 characters wouldn't avoid horizontal scrolling on my 15" MacBook Pro at 1440x900. On https://serviceworke.rs/push-simple_index_doc.html my browser displays only 69 characters of the code:

<img width="1440" alt="screen shot 2015-11-19 at 11 23 39" src="https://cloud.githubusercontent.com/assets/305455/11281683/73a857b6-8eb0-11e5-8bb3-32ce178fcbea.png">

(I can hide the sidebar, which works around the problem, but only until I reload the page.)

**Comment by marco-c** [comment]
On my laptop, I can see 101 characters.

**Comment by awalGarg** [comment]
@marco-c 

> It'd be great if you could also use the Q-style for promises, which makes lines shorter.

sure

@mykmelez 

> (I can hide the sidebar, which works around the problem, but only until I reload the page.)

hmm, we can use localstorage to remember state of sidebar across reloads? ;)
TBH, something below 80 chars sounds a wee bit too restrictive to me.

**Comment by digitarald** [comment]
Lets add a lint rule for 80 characters then.

**Comment by mykmelez** [comment]
> hmm, we can use localstorage to remember state of sidebar across reloads? ;)

That's a good idea!

> TBH, something below 80 chars sounds a wee bit too restrictive to me.

Indeed. Even 80 chars feels too limiting, even though well-formatted programs will rarely have lines that exceed that limit. It might be ok for a series of recipes, though.

**Comment by brendandahl** [comment]
We could also consider changing to an 'inline' style layout in docco.

**Comment by digitarald** [comment]
Throwing in 3 ideas that can be combined:

- make the docs column more narrow to save space, maybe even flexible to
some degree with flexbox
- make the code column overflow auto, so it can me scrolled
sideways separately from the docs on the right
- responsive layout that switches to inline style for narrow viewport and
expands to columns for wider views

On Thursday, November 19, 2015, Brendan Dahl <notifications@github.com>
wrote:

> We could also consider changing to an 'inline' style layout in docco.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/serviceworker-cookbook/issues/108#issuecomment-158242375>
> .
>


-- 
Web Craftsman | harald@mozilla.com | @digitarald [skype/irc/line] |
+1-650-963-6612 [wapp/telegram]

**Comment by awalGarg** [comment]
concur with @digitarald. A lint rule is probably the best idea.

I shall PR for storing nav status in localStorage and the lint rule+wrapping in a while.

Changing to the inline (officially called "linear", btw :)) style/responsiveness can be done in a separate issue, I guess?

**Comment by marco-c** [comment]
Fixed in #120 by @awalGarg.
